cfg/pre-commit-config.yaml

```diff
@@ -7,7 +8,75 @@ repos:
     entry: './src/sh/ensure-shebang.sh'
     id: 'ensure-shebang'
     language: 'script'
-    name: 'Ensure Shebang'
+    name: 'ensure shebang'
+    types:
+    - 'shell'
+  -
+    entry: './src/sh/python-lint/poetry-check.sh'
+    id: 'local-poetry-check'
+    language: 'script'
+    name: 'local poetry check'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run python -m file-name-conventions'
+      'Please rename the files to match the naming conventions.'
+    id: 'file-name-conventions'
+    language: 'script'
+    name: 'check file name conventions'
+    types:
+    - 'python'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run autoflake -c --remove-all-unused-imports
+      --remove-unused-variables --ignore-init-module-imports'
+      'Please run src/sh/python-lint/format.sh to fix the errors.'
+    id: 'autoflake'
+    language: 'script'
+    name: 'autoflake'
+    types:
+    - 'python'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run black --check'
+      'Please run src/sh/python-lint/format.sh to fix the errors.'
+    id: 'black'
+    language: 'script'
+    name: 'black'
+    types:
+    - 'python'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run flake8'
+      'Please fix the errors before committing.'
+    id: 'flake8'
+    language: 'script'
+    name: 'flake8'
+    types:
+    - 'python'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run isort --check'
+      'Please run src/sh/python-lint/format.sh to fix the errors.'
+    id: 'isort'
+    language: 'script'
+    name: 'isort'
+    types:
+    - 'python'
+  -
+    entry: |
+      './src/sh/python-lint/run-in-poetry-subdir.sh'
+      'poetry run mypy'
+      'Please fix the errors before committing.'
```

Comment:
This is inconsistent with the error message from the other hooks

Reply:
I'll explain more tomorrow, but the reason I made it different is because `mypy` just notifies you of errors, it doesn't actually change anything.

So running the `format.sh` script won't do anything except tell you the same errors again, you have to fix the errors yourself